specifyweb/frontend/js_src/lib/components/Header/index.tsx

```diff
@@ -194,9 +205,29 @@ export function MenuButton({
   readonly props?: TagProps<'a'> & TagProps<'button'>;
 }): JSX.Element | null {
   const [position] = userPreferences.use('header', 'appearance', 'position');
+  const [isSideBarLight] = userPreferences.use(
+    'general',
+    'appearance',
+    'lightSideBarBackground'
+  );
+  const isDarkMode = useDarkMode();
   const getClassName = (isActive: boolean): string => `
     p-[1.4vh]
-    ${isActive ? 'bg-brand-300 !text-white' : 'text-white'}
+    ${
+      isActive && isDarkMode
+        ? 'bg-brand-300 text-white'
+        : !isActive && isDarkMode
+        ? 'text-white'
+        : isActive && !isDarkMode && isSideBarLight
+        ? 'bg-brand-300 text-white'
+        : !isActive && !isDarkMode && isSideBarLight
+        ? 'text-gray-700'
+        : !isActive && !isDarkMode && !isSideBarLight
+        ? 'text-white'
+        : isActive && !isDarkMode && !isSideBarLight
+        ? 'bg-brand-300 text-white'
+        : 'text-white'
+    }
```

Comment:
this seems very overcomplicated.
your code is equivalent to this:
```ts
  isActive
    ? 'bg-brand-300 text-white'
    : isDarkMode || !isSideBarLight
    ? 'text-white'
    : 'text-gray-700';
```

(you can use a website like https://www.boolean-algebra.com/ or https://web.stanford.edu/class/cs103/tools/truth-table-tool/ to see how to simplify boolean expressions)

Reply:
so so much better